src/components/policy/policy_regular/src/policy_table/validation.cc

```diff
@@ -114,14 +114,46 @@ bool ApplicationPoliciesSection::Validate() const {
   return true;
 }
 
-bool ApplicationParams::Validate() const {
-  // Check for empty "groups" sub-sections
-  if (groups.empty()) {
-    return false;
+#ifdef SDL_REMOTE_CONTROL
+bool ApplicationParams::ValidateModuleTypes() const {
+  // moduleType is optional so see Optional<T>::is_valid()
+  bool is_initialized = moduleType->is_initialized();
+  if (!is_initialized) {
+    // valid if not initialized
+    return true;
+  }
+  bool is_valid = moduleType->is_valid();
+  if (is_valid) {
+    return true;
+  }
+
+  struct IsInvalid {
+    bool operator()(Enum<ModuleType> item) const {
+      return !item.is_valid();
+    }
+  };
+  // cut invalid items
+  moduleType->erase(
+      std::remove_if(moduleType->begin(), moduleType->end(), IsInvalid()),
+      moduleType->end());
+  bool empty = moduleType->empty();
+  if (empty) {
+    // set non initialized value
+    ModuleTypes non_initialized;
+    moduleType = Optional<ModuleTypes>(non_initialized);
   }
   return true;
 }
 
+bool ApplicationParams::Validate() const {
+  return ValidateModuleTypes();
```

Comment:
Is `if(groups.empty()) return false;` missing here?

Reply:
@JackLivio  why groups can't be empty?

Reply:
@JackLivio goups has type : 
```typedef Array<String<1, 255>, 0, 255> Strings;```
With minimal value - 1, so there no need for additional check of empty groups. It will be done implicitly.

Reply:
@JackLivio  why groups should not be empty?